content/usgs-packages/sbtools_Modify.Rmd

```diff
@@ -0,0 +1,302 @@
+
+```{r sbtools-update-mult}
+all_items <- item_list_children(user_id())
+update_contacts <- items_update(all_items, 
+             info = 
+               list(rep(list(contacts = list(list(name="Julius Caesar"))), 2)))
+
+# verify that the contacts were added to all items
+sapply(all_items, item_get_fields, fields="contacts")
+```
+
+If you want to update an item but it might not exist yet, use the `upsert` functions. Currently, this function is not working as you might expect. There are two `sbtools` issues that will address this: [issue 239](https://github.com/USGS-R/sbtools/issues/239) and [issue 240](https://github.com/USGS-R/sbtools/issues/240).
+
+*EXAMPLES NOT CURRENTLY WORKING*
+
+```{r sbtools-upsert, eval=FALSE}
+# creates a new item even though one already exists
+item_upsert(title="books.json") 
+
+# created another item with the name "books.json" with this message:
+#   "title is NULL - re-using title from input SB item"
+item_upsert(test_item, title=NULL, info=list(title = "sbtools stuff"))
+
+# created a new item named "sbtools stuff" under the 
+# books.json item
+item_upsert(test_item, title="sbtools stuff")
+```
+
+Another way to edit or update SB items is to add alternative identifiers, e.g. digital object identifiers, IPDS codes, etc. They can be useful to identify your items in searches because they have a user-defined type, scheme, and key. For examples of identifiers, see the "Additional Information | Identifiers" section of [Differential Heating](https://www.sciencebase.gov/catalog/item/580587a2e4b0824b2d1c1f23) (two sets of identifiers, one for a DOI and one for an IPDS, each of which has type/scheme/key) and [nwis_01645704](https://www.sciencebase.gov/catalog/item/556f2055e4b0d9246a9fc9f7) (one set of identifiers, created by us in our mda_streams scheme to help us search for items). Here we will show how to add alternative identifiers. See ?`query_item_identifier` if you want to know more about querying by alternative identifiers.
+
+Add identifiers to the item stored as the R object `test_item`. 
+
+```{r sbtools-identifier}
+item_update_identifier(test_item, scheme="example", type="sbtools", key="number 1")
+
+item_get(test_item)$identifiers # verify that identifiers exist now
+```
+
+## Editing and removing your files
+
+In addition to updating ScienceBase items, `sbtools` has functions to edit files. You can rename, replace, or remove files within SB items. 
+
+This section will give examples for the following functions: 
+
+* `item_rename_files`
+* `item_replace_files`
+* `item_rm_files`
+
+To rename files of a ScienceBase item, you need to provide a vector of the file names and a vector of new file names in the same order. Try renaming the `books.json` file to `bookinfo.json` using the `item_rename_files` function.
```

Comment:
are they really file paths? They should all be under the same item, so aren't you just giving the file names?